staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -371,6 +371,10 @@ func (o *ApplyOptions) Run() error {
 	visitedUids := sets.NewString()
 	visitedNamespaces := sets.NewString()
 
+	if o.Namespace != metav1.NamespaceNone {
```

Comment:
Let's use `if o.EnforceNamespace` which is a clear signal that user explicitly requested one here. Also please add a comment that this covers corner case, where the manifest was pointing to a single item which is being removed. Finally, please add a test case covering this.

Reply:
Okay, let me work on this.